### PR TITLE
[CLApp] Adding more tools to be able to print layer-wise info

### PR DESCRIPTION
**Pull request by AlejandroCornejo**
**📝 Description**
In this PR I am adding new capabilities to the Parallel Rule of Mixtures to be able to print several values (`DAMAGE`, `PLASTIC_DISSIPATION`, `PLASTIC_STRAIN_VECTOR` and `CAUCHY_STRESS_VECTOR`) at a desired layer (global axes).

This can be done now by writting (example with layer 1, could be up to 5):

`PLASTIC_STRAIN_VECTOR_LAYER_1`
`DAMAGE_LAYER_1`
`UNIAXIAL_STRESS_LAYER_1`
`CAUCHY_STRESS_VECTOR_LAYER_1`

I think that this will be really useful for you @balcayde

### CODE REVIEW

File: applications/ConstitutiveLawsApplication/constitutive_laws_application.cpp [hunk 1]
```diff
@@ -477,5 +479,32 @@ void KratosConstitutiveLawsApplication::Register()
     KRATOS_REGISTER_SYMMETRIC_3D_TENSOR_VARIABLE_WITH_COMPONENTS(ISOTROPIC_ANISOTROPIC_YIELD_RATIO);
     KRATOS_REGISTER_SYMMETRIC_3D_TENSOR_VARIABLE_WITH_COMPONENTS(ORTHOTROPIC_ELASTIC_CONSTANTS);
 
+    // For printing pursposes in Rule of Mixtures...
+    KRATOS_REGISTER_VARIABLE(CAUCHY_STRESS_VECTOR_LAYER_1)
```
Comment (loumalouomega) [inline applications/ConstitutiveLawsApplication/constitutive_laws_application.cpp:491]: Huuuuuum, I don't like this... Huuuum, thinking alternatives

File: applications/ConstitutiveLawsApplication/custom_constitutive/composites/rule_of_mixtures_law.h [hunk 2]
```diff
@@ -649,6 +649,17 @@ class KRATOS_API(CONSTITUTIVE_LAWS_APPLICATION) ParallelRuleOfMixturesLaw
...
 
+    /**
+     * @brief This method returns true if the variable is related to a layer
+     * @param rVariable The variable to be printed
+     */
+    template<class TVariableType>
+    const bool IsLayerVariable(const TVariableType& rVariable)
+    {
+        // returns true if LAYER is found
+        return (rVariable.Name().find("LAYER") != std::string::npos);
```
Comment (loumalouomega) [inline applications/ConstitutiveLawsApplication/custom_constitutive/composites/rule_of_mixtures_law.h:660]: String find is sloooow

**Review by loumalouomega — COMMENTED** [review]
Sorry I need to think a cleaner way to do this. @roigcarlo any suggestion?

File: applications/ConstitutiveLawsApplication/custom_constitutive/composites/rule_of_mixtures_law.h (hunk 2 above)
Comment (AlejandroCornejo) [reply applications/ConstitutiveLawsApplication/custom_constitutive/composites/rule_of_mixtures_law.h:660]: I could basically repeat a lot of if...else if with all the new variables but I though that in this way is easier to maintain

Comment (loumalouomega) [reply applications/ConstitutiveLawsApplication/custom_constitutive/composites/rule_of_mixtures_law.h:660]: Yes, I am thinking something else, like for example checking the variable Id if all variables of layer have a way to be filtered. Thinking. Luckily this is only post-process, but I need to think about it.

File: applications/ConstitutiveLawsApplication/constitutive_laws_application.cpp (hunk 1 above)
Comment (loumalouomega) [reply applications/ConstitutiveLawsApplication/constitutive_laws_application.cpp:491]: Maybe we can add a macro to add several variables at once, the declaration time will be the same but at least the code will be cleaner

**Comment by AlejandroCornejo** [comment]
ping

File: applications/ConstitutiveLawsApplication/constitutive_laws_application.cpp (hunk 1 above)
Comment (loumalouomega) [reply applications/ConstitutiveLawsApplication/constitutive_laws_application.cpp:491]: I am still not happy with this...

Comment (loumalouomega) [reply applications/ConstitutiveLawsApplication/constitutive_laws_application.cpp:491]: @pooyan-dadvand any idea how to solve it?

**Comment by AlejandroCornejo** [comment]
is there any alternative?

**Pull request closed without merging**